### PR TITLE
Graphical User Interface : design

### PR DESCRIPTION
**Pull request by anqitong**
#10

### CODE REVIEW

**Comment by anqitong** [comment]
Hint : download the picture to see it. Github does not display it entirely. For example, you don't see the content in the left rectangle : pencil, colors, text ...

**Comment by Thomas-dot-G** [comment]
No sure it is useful for now
And that is a really basic image, I mean, only the display, which is simple to change, not how it will be done on the interface.
(Usually, discussions are in issues)

**Comment by starsasumi** [comment]
I agree with Thomas. Maybe you can implement a static swing interface first.

P.S. I still can't see pencil, etc. Maybe you have not pull the last version?

**Comment by anqitong** [comment]
Then I'll simply implement a static swing it will be better instaed of svg picture

**Comment by Thomas-dot-G** [comment]
@anqitong Are you still working on the code ? (Because you are still assigned and nothing changed in 4 days, but it can be normal as it is not urgent, just asking)

**Comment by anqitong** [comment]
oui c'est normal je continue de travailler là-dessus

**Comment by anqitong** [comment]
What do you think if we add some text fields for staff such as Circle : it would need for example x, y and radius. So that the user would only need to type the values and validate. 
For the color picker, shall we use the ColorDialog or simply let the user enter the rgb value?

**Comment by Thomas-dot-G** [comment]
I think RGB is fine for now (the ColorDialog would be a new feature if we really have time).
For the Field, I agree, but make them Integer Fields

**Comment by anqitong** [comment]
All right, I also thought that color dialog is not yet necessary yet

**Comment by anqitong** [comment]
Shall I add the listeners in this PR too?

**Comment by Thomas-dot-G** [comment]
As you wish. We can do that separately if that works (haven't tested it yet).

But just, why have you added an image of a tiger ??

**Comment by anqitong** [comment]
By listeners I mean at least preparing the classes.

**Comment by Thomas-dot-G** [comment]
Yeah

**Comment by anqitong** [comment]
To display a svg file in swing, I will need to use this library : http://thinktibits.blogspot.fr/2012/12/Batik-Convert-SVG-PNG-Java-Program-Example.html
Otherwise swing is not able to display svg. That is why I used the tiger image that is a png format.

If you're not against this library, let me know.

**Comment by Thomas-dot-G** [comment]
Why do you want to display images ? That is not the goal, and for the awt, @starsasumi is doing it.

I thought the interface was just for the user to say, I want a circle, enter the parameters, then add that, with that pencil...
ie, replace the main function by what you are doing.

**Comment by anqitong** [comment]
I'm not doing the awt, it will draw nothing but the svg through xml.

**Comment by anqitong** [comment]
It would be nice to enter the parameters , create the svg, and then display the svg in the window

**Comment by Thomas-dot-G** [comment]
Yes, but I didn't understand your problem with the import of your image then

**Comment by anqitong** [comment]
Swing is not able to display a svg, it needs some libraries to do it.

**Comment by anqitong** [comment]
@Thomas-dot-G : can you have a look at the design? I would like to finish this branch and merge it so that I can start the controllers in another PR.

**Comment by Thomas-dot-G** [comment]
Ok I will, I let you know when.
But I still don't get why we need that picture of a tiger in the git...

**Comment by anqitong** [comment]
I'll take it away soon.

File: files/docs/guiDesign.svg [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+       transform="translate(-1.4285714,509.69629)"><flowRegion
+         id="flowRegion4198"><rect
+           id="rect4200"
+           width="134.28571"
+           height="169.99998"
+           x="14.285714"
+           y="362.66592"
+           style="font-size:25px" /></flowRegion><flowPara
+         id="flowPara4202">Colors</flowPara></flowRoot>  </g>
+</svg>
```
Comment (Thomas-dot-G) [inline files/docs/guiDesign.svg]: I think you can remove that

**Comment by Thomas-dot-G** [comment]
Nice, but I would have put a radio box for different shapes (for the user to only select one at the time) and then when one is selected, it shows the parameters to enter with a "add" button (that will in the future add the command line on the left)
(But I guess with warning, it can stay like that if we decide that clicking on the name of the shape does mean "add" if the shape has been added.

You can quickly also add color for the closed shapes

For polylines and polygone we can see that later if you want

**Comment by starsasumi** [comment]
The gui is fine with me too. 

One thing related to the GUI that worth to mention (again) is that a java software can not run a java source code directly (issue #19). As the result, the run button would not work as a trigger to generate the SVG and the AWT according to the script (which is a java source code).

Or... the run button will just create the source file and the user will run `javac ...` him/herself?

**Comment by Thomas-dot-G** [comment]
Still don't get why @starsasumi .
Clicking on a generate button can absolutely write some text in a file, and thus create a svg file.
I've already done that in several projects for saving the state of the software with some backups (or saving for game software). So while running, the software can well create a svg file.
The run button can also open a new window (can be done in awt also, already tried)

**Comment by starsasumi** [comment]
@Thomas-dot-G No no, what I mean is we can not generate images from the script, which is required in the task I think.

`A script (or high level description language) -> middle level description language -> presentations`

**Comment by Thomas-dot-G** [comment]
What do you mean we can not generate images from the script ?
As far as I know, our "script" can generate images, and the GUI will not prevent us from doing so...

**Comment by starsasumi** [comment]
The script is a source code (`.java`), it's not a compiled `.class` which could be loaded and run by Java Runtime. To run a `.java`, typically, it need to go through compiler and become `.class`

**Comment by starsasumi** [comment]
If we load the script at runtime, we can either:
1. do the [dynamically compile](http://www.beyondlinux.com/2011/07/20/3-steps-to-dynamically-compile-instantiate-and-run-a-java-class/)
2. analyze the script ourselves, which is technically implement a small and special compiler.

**Comment by Thomas-dot-G** [comment]
That is because you want to add a special script, isn't it ?
But with the Gui, add a circle will just call the function add (in the ArrayList<Shapes> created at the beginning(and empty)) the circle. We'll need to add new methods, but I can be done without creating a special compiler. Just controllers and some methods are needed.
In the main of the GUI, the two different SVG and AWT object would be initialized empty, and one of them (or both) will be filled with the GUI, then displayed.
I think we can easily overcome that small obstacle that is no dynamic compiling.

**Comment by anqitong** [comment]
Good idea with radio button I'll change to it

**Comment by starsasumi** [comment]
I do understand we can use new methods to create images with our GUI. But don't we need to use the script language that we created since it's a part of the assignment? Maybe we should ask the prof. for this.

**Comment by anqitong** [comment]
@starsasumi : where exactly in the subject do you see that we have to use the script language?

**Comment by starsasumi** [comment]
@anqitong here: http://www.grall.name/teaching/softwareArchitecture/2016/project_vectorGraphics.html

```
L'objectif de ce projet est triple :

 - la définition d'un langage simple pour dessiner logiquement,
 - la définition d'interprétations des dessins logiques pour les afficher dans des espaces physiques,
 - l'interfaçage du langage de dessin avec Java, permettant d'écrire directement un script de dessin en Java.
```

Maybe it's just me who understand it wrong.

**Comment by starsasumi** [comment]
In case I didn't say it clear, I am OK to merge this PR and discuss the script problem in other place (like a new issue)

**Comment by Thomas-dot-G** [comment]
Ok, but as the teacher said, the "script" can just be the main, it's how we create the svg for a user, that's all

**Comment by anqitong** [comment]
Im' not done yet, I need to change to radio buttons

**Comment by anqitong** [comment]
@Thomas-dot-G : I also wanted to hide/show the fields under the different shapes when the user clicks on it, but that should be done in the listeners.

**Comment by Thomas-dot-G** [comment]
Ok then

**Comment by Thomas-dot-G** [comment]
Add a button "add", to add the shape (once one shape done, we might want to add it and then clear the left side to add another one)

**Comment by Thomas-dot-G** [comment]
your last commit says remove ...svg, but it wasn't done

**Comment by anqitong** [comment]
Strange, it is indeed deleted in my project : guiDesign.svg

**Comment by anqitong** [comment]
It wasn't committed that's why ...

**Comment by Thomas-dot-G** [comment]
ok, would you mind doing it then ?

**Comment by anqitong** [comment]
yes it is already done, go check the last commit

**Comment by Thomas-dot-G** [comment]
I would have put that add button at the end, but that is fine.
(After filling the fields, that's when you want to add the shape.)
If you can move it, I think it will be ready to merge

**Comment by anqitong** [comment]
Can I merge?

**Pull request merged**